src/field_special_scene.c

```diff
@@ -0,0 +1,365 @@
+       sub_808E82C(3, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, -3 - cameraXpan, cameraYpan);
+   }
+}
+
+void Task_HandleTruckSequence(u8 taskId)
+{
+   s16 *data = gTasks[taskId].data;
+
+    switch (data[0])
+    {
+        /*
+        Each case has a timer which is handled with data[1], incrementing
+        until it reaches the if function's condition, which sets the next task up.
+        */
+    case 0:
+        data[1]++;
+        if (data[1] == SECONDS(1.5))
+        {
+            SetCameraPanningCallback(NULL);
+            data[1] = 0; // reset the timer.
+            data[2] = CreateTask(Task_Truck1, 0xA);
+            data[0] = 1; // run the next case.
+            PlaySE(SE_TRACK_MOVE);
+        }
+        break;
+    case 1:
+        data[1]++;
+        if (data[1] == SECONDS(2.5))
+        {
+            pal_fill_black();
+            data[1] = 0;
+            data[0] = 2;
+        }
+        break;
+    case 2:
+        data[1]++;
+        if (!gPaletteFade.active && data[1] > SECONDS(5))
+        {
+            data[1] = 0;
+            DestroyTask(data[2]);
+            data[3] = CreateTask(Task_Truck2, 0xA);
+            data[0] = 3;
+            PlaySE(SE_TRACK_STOP);
+        }
+        break;
+    case 3:
+        if (!gTasks[data[3]].isActive) // is Truck2 no longer active (is Truck3 active?)
+        {
+            InstallCameraPanAheadCallback();
+            data[1] = 0;
+            data[0] = 4;
+        }
+        break;
+    case 4:
+        data[1]++;
+        if (data[1] == 90)
+        {
+            PlaySE(SE_TRACK_HAIK);
+            data[1] = 0;
+            data[0] = 5;
+        }
+        break;
+    case 5:
+        data[1]++;
+        if (data[1] == 120)
+        {
+            MapGridSetMetatileIdAt(11, 8, 520);
+            MapGridSetMetatileIdAt(11, 9, 528);
+            MapGridSetMetatileIdAt(11, 10, 536);
+            DrawWholeMapView();
+            PlaySE(SE_TRACK_DOOR);
+            DestroyTask(taskId);
+            ScriptContext2_Disable();
+        }
+        break;
+    }
+}
+
+void ExecuteTruckSequence(void)
+{
+    MapGridSetMetatileIdAt(11, 8, 525);
+    MapGridSetMetatileIdAt(11, 9, 533);
+    MapGridSetMetatileIdAt(11, 10, 541);
+    DrawWholeMapView();
+    ScriptContext2_Enable();
+    CpuFastFill(0, gPlttBufferFaded, 0x400);
+    CreateTask(Task_HandleTruckSequence, 0xA);
+}
+
+void EndTruckSequence(u8 taskId)
+{
+    if (!FuncIsActiveTask(Task_HandleTruckSequence))
+    {
+        sub_808E82C(1, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, 3, 3);
+        sub_808E82C(2, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, 0, -3);
+        sub_808E82C(3, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, -3, 0);
+    }
+}
+
+extern u8 GetSSTidalLocation(s8 *, s8 *, s16 *, s16 *); // should be in field_specials.h
```

Comment:
Any reason why you haven't created that already then? These externs would be cleaner in a shared header.

Reply:
Because it would be an extern as well since the function is a part of field_specials.c, but that file isnt decompiled yet.

Reply:
Fair enough. As a matter of tidiness, they should still probably not be down here scattered in between functions; you should probably move them to the top of the file, at least until the proper header file is created.